contracts/claimdrops.js

```diff
@@ -0,0 +1,346 @@
+  if (tableExists === false) {
+    await api.db.createTable('claimdrops', ['claimdropId', 'symbol']);
+    await api.db.createTable('params');
+
+    const params = {};
+    params.creationFee = '50';
+    params.feePerClaim = '0.1';
+    // 90 days (in milliseconds)
+    params.maxExpiryTime = 7776000000;
+    params.maxExpiresPerBlock = 5;
+    await api.db.insert('params', params);
+  }
+};
+
+actions.updateParams = async (payload) => {
+  if (api.assert(api.sender === api.owner, 'not authorized')) {
+    const {
+      creationFee,
+      feePerClaim,
+      maxExpiryTime,
+      maxExpiresPerBlock,
+    } = payload;
+
+    const params = await api.db.findOne('params', {});
+
+    if (creationFee) {
+      if (!api.assert(typeof creationFee === 'string' && !api.BigNumber(creationFee).isNaN() && api.BigNumber(creationFee).gte(0), 'invalid creationFee')) return;
+      params.creationFee = creationFee;
+    }
+    if (feePerClaim) {
+      if (!api.assert(typeof feePerClaim === 'string' && !api.BigNumber(feePerClaim).isNaN() && api.BigNumber(feePerClaim).gte(0), 'invalid feePerClaim')) return;
+      params.feePerClaim = feePerClaim;
+    }
+    if (maxExpiryTime) {
+      if (!api.assert(Number.isInteger(maxExpiryTime) && maxExpiryTime > 0, 'invalid maxExpiryTime')) return;
+      params.maxExpiryTime = maxExpiryTime;
+    }
+    if (maxExpiresPerBlock) {
+      if (!api.assert(Number.isInteger(maxExpiresPerBlock) && maxExpiresPerBlock > 0 && maxExpiresPerBlock <= 1000, 'invalid maxExpiresPerBlock')) return;
+      params.maxExpiresPerBlock = maxExpiresPerBlock;
+    }
+
+    await api.db.update('params', params);
+  }
+};
+
+const hasValidPrecision = (value, precision) => (api.BigNumber(value).dp() <= precision);
+
+const transferIsSuccessful = (result, action, from, to, symbol, quantity) => {
+  if (result.errors === undefined
+    && result.events && result.events.find(el => el.contract === 'tokens'
+    && el.event === action
+    && el.data.from === from
+    && el.data.to === to
+    && api.BigNumber(el.data.quantity).eq(quantity)
+    && el.data.symbol === symbol) !== undefined) {
+    return true;
+  }
+
+  return false;
+};
+
+const getTimestamp = (value) => {
+  try {
+    const date = new Date(`${value}.00Z`);
+    return date.getTime();
+  } catch (e) {
+    return false;
+  }
+};
+
+const validateList = (list, precision) => {
+  const parsedList = [];
+  for (let i = 0; i < list.length; i += 1) {
+    const { 0: account, 1: limit } = list[i];
+
+    // account & limit validation
+    if (api.assert(account, `list[${i}]: account name cannot be undefined`)
+      && api.assert(api.isValidAccountName(account), `list[${i}]: invalid account name`)
+      && api.assert(limit, `list[${i}]: limit cannot be undefined`)
+      && api.assert(!api.BigNumber(limit).isNaN(), `list[${i}]: invalid limit`)
+      && api.assert(api.BigNumber(limit).gt(0), `list[${i}]: limit must be positive`)
```

Comment:
Can you do a max size check for limit?  Can be something like asserting that api.BigNumber(limit).lte(Number.MAX_SAFE_INTEGER) and also use the BigNumber isFinite function - https://mikemcl.github.io/bignumber.js/#isF